Source/WebKit/Shared/WebHitTestResultData.h

```diff
@@ -84,6 +84,8 @@ struct WebHitTestResultData {
     bool isTextNode;
     bool isOverTextInsideFormControlElement;
     bool isDownloadableMedia;
+    bool hasMediaElement;
+    bool mediaIsVideo;
```

Comment:
Can we make the implementation match the SPI surface – using an enumeration rather than two booleans?

We may want to extend this in the future. Something that comes to mind is a way to expose that a `<model>` element is being context clicked.

Reply:
Should `_WKHitTestResultMediaType` and `mediaType` names be more generic in `_WKHitTestResult`? Maybe `_WKHitTestResultElementType` and `elementType`?

Reply:
That sounds good to me!